src/notifications/common.ts

```diff
@@ -481,18 +481,26 @@ export const generateUserNotificationUniqueKey = ({
   type,
   referenceId,
   referenceType,
+  dedupKey,
 }: {
   type: NotificationType;
   referenceId?: string;
   referenceType?: NotificationReferenceType;
+  dedupKey?: string;
 }): string | null => {
   const uniqueKey = notificationTypeToUniqueKey[type];
 
   if (!uniqueKey) {
     return null;
   }
 
-  return [uniqueKey, referenceId, referenceType].filter(Boolean).join(':');
+  return [
+    uniqueKey,
+    dedupKey ? `dedup_${dedupKey}` : referenceId,
+    referenceType,
```

Comment:
yeah think this works now, basically (correct me if I am wrong):
- if dedup key is present we send only single notification for that post
- if no dedup key we dedup per postId (as before)